Video: addition of the imx335 driver (#88825)
zephyrproject-rtos/zephyr: boards/shields/st_b_cams_imx_mb1854/doc/index.rst

.. _st_b_cams_imx_mb1854:

ST B-CAMS-IMX-MB1854
####################

Overview
********

The B-CAMS-IMX camera module provides a compelling hardware set to
handle multiple computer vision scenarios and use cases. It features
a high-resolution 5‑Mpx RGB CMOS image sensor, an inertial motion unit,
and a Time‑of‑Flight sensor. It can be used with any STM32 boards featuring
a MIPI CSI-2® interface with a 22‑pin FFC connector to enable full-featured
computer vision on STM32 microcontrollers and microprocessors easily.

.. figure:: st_b_cams_imx.webp
     :width: 600px
     :align: center
     :alt: B-CAMS-IMX-MB1854

     B-CAMS-IMX MB1854 Image (Credit: STMicroelectronics.)

Requirements
************

The camera module bundle is compatible with all STM32 Discovery kits and
Evaluation boards featuring a 22 pins FFC connector, such as the STM32N6570_DK
Discovery kit.

Usage
*****

The shield can be used in any application by setting ``SHIELD`` to

> ``_st_b_cams_imx_mb1854`` and adding the necessary device tree properties.

Reviewer: [non-blocking]

> and adding the necessary device tree properties

Is that not up to the shield to provide a [`shields/<shieldname>/boards/<boardname>.overlay`](https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/shields/x_nucleo_bnrg2a1/boards) in case there is some board-specific integration needed? Ideally it is not needed and the shields works drop-in thanks to standardized nodelabels though.

Possible alternative wording to avoid ambiguity:

```suggestion
``_st_b_cams_imx_mb1854`` for boards with the necessary device tree node labels.
```

Reviewer: Is the leading `_` in `_st_b_cams_imx_mb1854` a typo? It is not present below.

Author: Yes indeed, I remove the "adding the necessary device tree properties" part. and did the other updates / typo fix.
